app/models/piece.rb

```diff
@@ -1,3 +1,102 @@
 class Piece < ApplicationRecord
   belongs_to :game
+  validates :color, presence: true
+  validates :type, presence: true
+
+  scope :white, ->() { where(color: 'white') }
+  scope :black, ->() { where(color: 'black') }
+
+  def self.at(x_position, y_position)
+    Piece.find_by(x_position: x_position, y_position: y_position)
```

Comment:
Since we can have multiple games in our database, this needs to check for the correct game as well.

Otherwise, this could return a piece that's in another game.